model/src/form/form-definition/types.ts

```diff
@@ -29,6 +43,14 @@ export interface PageQuestion extends PageBase {
   components: ComponentDef[]
 }
 
+export interface PageRepeat extends PageBase {
```

Comment:
Is this why our type narrowing isn't working?

Because a `PageQuestion` can't be a `PageRepeat`—they're separate interfaces